akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/javadsl/TestProbe.scala

```diff
@@ -65,7 +67,7 @@ object TestProbe {
  * Not for user extension
  */
 @DoNotInherit
-abstract class TestProbe[M] {
+abstract class TestProbe[M] extends RecipientRef[M] { this: InternalRecipientRef[M] =>
```

Comment:
I'm assuming that the self-type doesn't mess with the `javadsl`'s Java-ness.

Reply:
The self type is impl internal and as this is not for inheriting from user code it should be fine